sdcm/nemesis.py

```diff
@@ -434,7 +438,123 @@ def disrupt_stop_start_scylla_server(self):  # pylint: disable=invalid-name
         self.target_node.stop_scylla_server(verify_up=False, verify_down=True)
         self.target_node.start_scylla_server(verify_up=True, verify_down=False)
 
+    def disrupt_start_stop_major_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 120
+        trigger_func = partial(compaction_ops.trigger_major_compaction)
+        watch_func = partial(compaction_ops.stop_on_user_compaction_logged,
+                             node=node,
+                             watch_for="User initiated compaction",
+                             timeout=timeout,
+                             stop_func=compaction_ops.stop_major_compaction)
+        self.tester.wait_no_compactions_running()
+        ParallelObject(objects=[trigger_func, watch_func], timeout=timeout).call_objects()
+
+    def disrupt_start_stop_scrub_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 120
+        trigger_func = partial(compaction_ops.trigger_scrub_compaction)
+        watch_func = partial(compaction_ops.stop_on_user_compaction_logged,
+                             node=node,
+                             watch_for="Scrubbing",
+                             timeout=timeout,
+                             stop_func=compaction_ops.stop_scrub_compaction)
+        self.tester.wait_no_compactions_running()
+        ParallelObject(objects=[trigger_func, watch_func], timeout=timeout).call_objects()
+
+    def disrupt_start_stop_cleanup_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 120
+        trigger_func = partial(compaction_ops.trigger_cleanup_compaction)
+        watch_func = partial(compaction_ops.stop_on_user_compaction_logged,
+                             node=node,
+                             timeout=timeout,
+                             watch_for="Cleaning",
+                             stop_func=compaction_ops.stop_cleanup_compaction)
+        self.tester.wait_no_compactions_running()
+        ParallelObject(objects=[trigger_func, watch_func], timeout=timeout).call_objects()
+
+    def disrupt_start_stop_validation_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        timeout = 120
+        trigger_func = partial(compaction_ops.trigger_validation_compaction)
+        watch_func = partial(compaction_ops.stop_on_user_compaction_logged,
+                             node=node,
+                             watch_for="Scrubbing in validate mode",
+                             timeout=timeout,
+                             stop_func=compaction_ops.stop_validation_compaction)
+        self.tester.wait_no_compactions_running()
+        ParallelObject(objects=[trigger_func, watch_func], timeout=timeout).call_objects()
+
+    def disrupt_start_stop_upgrade_compaction(self):
```

Comment:
let's move it to the test since we don't support upgrades during the longevities with Nemesis

Reply:
Fixed (moved back to the test module)